packages/kit/types/index.d.ts

```diff
@@ -2062,16 +2125,14 @@ declare module '$app/paths' {
 	 * ); // `/blog/hello-world/something/else`
 	 * ```
 	 * */
-	export function resolveRoute(id: string, params: Record<string, string | undefined>): string;
+	export function resolveRoute<K extends RouteWithParams>(id: K, params: RouteIds[K]): string;
+	export function resolveRoute<K extends RouteWithoutParams>(id: K): string;
 }
```

Comment:
Lint is not passing because I wrote the type here directly. It seems that we have to do it in `JSDoc`, but I don't manage to do it.
I would need some guidance.

Reply:
I too am somewhat flummoxed by JSDoc in this case. For now I've put this stuff in a src `.d.ts` file, which sort of works, though there are some other problems that need solving separately.

I took the liberty of moving the `RouteId` type to `$types` rather than `$app/paths` — it's arguably more consistent with the various `./$types` files, and it'll be convenient to have a single place to put all the generated types.

Right now it doesn't seem like that ambient declaration is getting picked up — working on it...